WalletWasabi/WabiSabi/Client/HttpRequestHandlerProxy.cs

```diff
@@ -0,0 +1,70 @@
+using Newtonsoft.Json;
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using WalletWasabi.Tor.Http;
+using WalletWasabi.Tor.Http.Extensions;
+using WalletWasabi.WabiSabi.Backend.PostRequests;
+using WalletWasabi.WabiSabi.Models;
+
+namespace WalletWasabi.WabiSabi.Client
+{
+	public class HttpRequestHandlerProxy : IArenaRequestHandler
+	{
+		private IHttpClient _client;
+
+		public HttpRequestHandlerProxy(IHttpClient client)
```

Comment:
How will we get different identities. Do we need the `HttpClientFactory`?

Reply:
i am still thinking. An individual Alice can reuse the same stream for registering input and confirm connection because it needs to identify herself with her aliceID so, Alice is easy. Bob needs to use a different stream than Alice to break the link between them  what shouldn't be a problem. 

What seems to be a problem is that the TorHttpClient doesn't support our needs because if we specify `isolateStream=true` then every request uses a new stream (this is not good for our Alice) and if we `isolateStream=false` then it will reuse connections so, it shoud have it own connection pool  with only one connection but this is not possible neither.

Reply:
@kiminuo do you have an idea for this?

Reply:
> What seems to be a problem is that the TorHttpClient doesn't support our needs because if we specify `isolateStream=true` then every request uses a new stream (this is not good for our Alice) and if we `isolateStream=false` then it will reuse connections so, it shoud have it own connection pool with only one connection but this is not possible neither.

I removed this functionality from `TorHttpClient` and `IHttpClient` as you did not like it. :) I think it can be added back and it should not be that difficult.

Reply:
I am not sure, i need to play a bit with this first in order to understand better.

Reply:
I think we can handle this later, but what I am hoping for in the long run:

1. maintain a pool of connections to the coordinator on isolated circuits. new connections should be created randomly, and connections should have a random time to live that is not too high (on the order of minutes).
2. randomly perform non identifying requests (e.g. fetching round metadata) on these connections, using HTTP keep alive. there are some nuances WRT how to do this privately (connections should be treated as independent) and efficiently, but for an initial version i think we can just use a fixed interval for each connection and avoid anything fancy like long polling or streamed updates or whatever
3. when the credential schedule is computed, the total number of separate connections will be needed, the rate of new circuit creation can be increased so that the average number of open circuits is sufficient.
4. when an `ArenaClient` need a connection (initial or after a failed request), it can first check the pool, filtering only good connections (below some latency threshold) and "steal" a connection from the pool. if it's an output registration or reissuance, the arena client will only be used for one request and the connection should be disposed immediately, but arena clients used by `Alice` clients can make multiple requests on the same connection with no degradation of privacy before disposing the connection.

So this would needs something in between the current behaviour of `isolateStream=true` and `isolateStream=false`.

However, let's not worry about it right now, for testing we can just avoid the tor client altogether, and for MVP with privacy we can use `isolateStream=true` and hope the tor daemon keeps enough circuits already built so that establishing connections is not too slow or unreliable.

Reply:
@kiminuo it was this change https://github.com/zkSNACKs/WalletWasabi/pull/4767 with the `DefaultIsolateStream`. It would not solve the problem IMO. Bob, we need different streams `isolateStream=true` is good, every call is on a different stream. For Alice we need a dedicated stream initially and every call should go on that - `isolateStream=false` won't do that, correct me if I am wrong.

Reply:
For Alice we don't strictly need that, it's an optimization: requests can still be isolated and everything should work correctly, it's just very wasteful to do that since Tor would have to create one circuit per request, of which there can be many, but isolating in this way provides no privacy benefit because the requests of a single Alice are inherently linkable to each other.

Reply:
@molnard 
> @kiminuo it was this change #4767 with the `DefaultIsolateStream`. It would not solve the problem IMO.

My comment was a bit tongue in cheek. Anyway, it seems helpful to have more control over which tor circuit is used for an HTTP request and that does not seem to be that hard to achieve (an ugly way may be simply to add "identity" identifier of  a circuit or possibly some context object containing more nuanced info about how to send the request using Tor).